marconi-chain-index/src/Marconi/ChainIndex/Experimental/Runner.hs

```diff
@@ -56,27 +62,81 @@ runIndexer
   -> indexer (WithDistance BlockEvent)
   -> IO ()
 runIndexer socketPath networkId _startingPoint traceName indexer = do
```

Comment:
As we have explicit export now in the module, maybe we can avoid putting everything in the where clause, and try to split it into smaller groups of cohesive functions, to improve readability. Even with small functions, a where clause of 80 lines is big.

Reply:
Moved out the `readEvent` function.